operator/data/examples/multicluster/values-istio-multicluster-primary.yaml

```diff
@@ -14,8 +14,7 @@ spec:
       meshNetworks:
         network1:
           endpoints:
-          # Always use Kubernetes as the registry name for the main cluster in the mesh network configuration
-          - fromRegistry: Kubernetes
+          - fromRegistry: main0
```

Comment:
Shouldn't we change L9 to `Kubernetes`

Reply:
I don't think so, the user is free to name their clusters however they choose. The issue seems to be that we don't treat the clusterID consistently throughout the code.

Reply:
But we have `Kubernetes` as the default clusterID.  I don't think only change it here can work well

Reply:
@hzxuzhonghu can you, expand on why this might not work, specifically? I think that's what we're trying to understand ... this "seems" like the right change, but we don't fully understand all of the moving parts here.

Reply:
First we have env `CLUSTER_ID` here:
https://github.com/istio/istio/blob/327d9b5afb8d57b12692ed3da3f1329db1130575/pilot/pkg/features/pilot.go#L298-L299

And we have args `clusterID` for istiod, which default to the above env if not provided. 

The change here must combine with the change of these values as well.

Reply:
Those will be overridden in any usecase that we specify clusterName, which we would here.